Userland/Libraries/LibWeb/Loader/ResourceLoader.h

```diff
@@ -98,6 +99,7 @@ class ResourceLoader : public Core::Object {
     HashTable<NonnullRefPtr<ResourceLoaderConnectorRequest>> m_active_requests;
     NonnullRefPtr<ResourceLoaderConnector> m_connector;
     String m_user_agent;
+    NonnullRefPtr<Core::Timer> m_timer = Core::Timer::create_single_shot(0, [] {});
```

Comment:
hmm, perhaps we could make this a RefPtr and only create a timer if the request's timeout is actually set?